doc/adr/0013-synchronous_peer_discovery.md

```diff
@@ -0,0 +1,136 @@
+# 14. Synchronous Peer Discovery
+
+Date: 2023-04-19
+
+## Status
+
+Proposed
+
+## Context
+
+In order to increase security, it is desirable to observe only peers that are on the majority fork.
+This can be determined by querying discovered peers for their latest ordinal values prior to
+selecting one to observe. Afterwards, the majority ordinal can be determined and each peer is queried
+again for the `Signed[GlobalSnapshot]` corresponding to that ordinal. The hashes of the snapshots
+are compared to determine the most prevelant hash among the peers. A peer with that snapshot is then
+randomly chosen to be the one to be observed.
+
+Right now, the peer discovery process is a non-terminating process that runs alongside the main
+process. It listens to and processes peers that are put in a queue. Therefore, there is only 1 known
+peer at the start of observation 99% of the time. In other words, determining the majority fork from
+a list of 1 is not useful. So, it is imperative that there be multiple peers to choose from when
+choosing one to observe.
+
```

Comment:
It makes sense to me to stop here. Later on we propose a solution that does not change joining or peer discovery and still address the concern above.

Reply:
Addressed.